compiler/rustc_pattern_analysis/src/constructor.rs

```diff
@@ -266,6 +266,7 @@ pub struct IntRange {
 
 impl IntRange {
     /// Best effort; will not know that e.g. `255u8..` is a singleton.
+    #[cfg_attr(not(feature = "rustc"), allow(dead_code))]
```

Comment:
Why is this gated?

Reply:
I wanted it to compile without warnings with either feature, I could also just make it `pub`

Reply:
Well if it's unused with `not(feature = "rustc")`, then it's actually unused, and shouldn't we just cfg it out? If you expect it to be public for things like r-a, then make it pub.